homeassistant/components/esphome/climate.py

```diff
@@ -285,7 +275,9 @@ async def async_set_temperature(self, **kwargs: float | str) -> None:
         """Set new target temperature (and operation mode if set)."""
         data: dict[str, Any] = {"key": self._static_info.key}
         if ATTR_HVAC_MODE in kwargs:
-            data["mode"] = _CLIMATE_MODES.from_hass(cast(str, kwargs[ATTR_HVAC_MODE]))
+            data["mode"] = _CLIMATE_MODES.from_hass(
+                cast(HVACMode, kwargs[ATTR_HVAC_MODE])
```

Comment:
Why is `kwargs` typed as `**kwargs: float | str`, but we cast it to `HVACMode`?
Note that the `kwargs` are not type in the base component.

Reply:
I have adjusted to set `**kwargs` as `Any`